Add auth support, split-by-type option, and pagination for large graphs

shahar-biron opened the pull request:
## Changes
- Add --username and --password options for FalkorDB authentication
- Add --split-by-type/--no-split-by-type flags to control CSV file organization
- Implement pagination to export all nodes and edges (fixes 10k limit issue)
- Add progress feedback during export

## Problem Solved
The exporter was previously limited to 10,000 nodes/edges due to FalkorDB's default result limit. This PR adds pagination to fetch all data in batches.

## New Features
- **Authentication**: Optional username/password for secure FalkorDB connections
- **Flexible output**: Choose between separate files per type or combined files
- **Progress tracking**: Real-time feedback on export progress

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Optional authentication for exports and CLI flags for username/password and split-by-type
  * Configurable output: per-label/type CSVs (default) or consolidated nodes.csv/edges.csv
  * Batched, paginated exports for large graphs with per-label/type progress logging and final summary counts

* **Documentation**
  * README and CLI docs updated with auth, pagination, split/combined examples

* **Tests**
  * Added tests for auth, combined-mode, default split-mode, and pagination on large graphs
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

shahar-biron committed:
Add auth support, split-by-type option, and pagination for large graphs

- Add --username and --password options for FalkorDB authentication
- Add --split-by-type/--no-split-by-type flags to control CSV file organization
- Implement pagination to export all nodes and edges (fixes 10k limit)
- Add progress feedback during export

shahar-biron committed:
Update README with new command line options and usage examples

- Document authentication options (--username, --password)
- Explain --split-by-type and --no-split-by-type modes
- Add comprehensive usage examples
- Document pagination support for large graphs
- Update output section with file naming patterns

shahar-biron committed:
Add tests for new features to improve code coverage

- Test authentication parameter acceptance
- Test split-by-type mode (both True and False)
- Test combined mode output (single nodes.csv and edges.csv)
- Test default behavior (split mode)
- Test pagination with larger graphs (100 nodes)

AviAvni commented on main.py:
```diff
@@ -4,64 +4,130 @@
 import pandas as pd
 
-def export_graph(graph_name, host, port):
+def export_graph(graph_name, host, port, username=None, password=None, split_by_type=True):
```
use url instead

AviAvni commented on main.py:
```diff
@@ -4,64 +4,130 @@
+
+    while True:
+        nodes_result = g.ro_query(
+            f"MATCH (n) RETURN ID(n), labels(n), properties(n) SKIP {skip} LIMIT {batch_size}"
```
It will be faster if instead of skip filter wit id(n) > last id

shahar-biron replied:
id is not necessarily sortable... I think current approach is "good enough"...

AviAvni replied:
this query specifically under the hood is scan the nodes by id
it will be slow when the graph is very large

shahar-biron committed:
Address AviAvni review comments: use URL connection and ID-based pagination

- Use redis:// URL format for connection instead of separate parameters
- Replace SKIP-based pagination with ID-based filtering (WHERE ID(n) > last_id)
- Add ORDER BY ID(n)/ID(e) for consistent pagination
- Improves performance on large graphs by avoiding expensive SKIP operations

shahar-biron committed:
Revert to SKIP pagination - FalkorDB doesn't support ORDER BY ID()

The ID-based pagination approach failed because FalkorDB/Cypher doesn't support
ORDER BY on ID() function. Reverting to SKIP-based pagination while keeping the
URL connection improvement from AviAvni's review.

shahar-biron committed:
Fix test_export_connection_error to use reliable connection failure

shahar-biron committed:
Remove test_export_connection_error - connection behavior not guaranteed to raise
